indexing-service/src/main/java/org/apache/druid/indexing/common/task/KillUnusedSegmentsTask.java

```diff
@@ -237,14 +237,26 @@ public TaskStatus runTask(TaskToolbox toolbox) throws Exception
       // the metadata segment is present. If the segment nuke throws an exception, then the segment cleanup is
       // abandoned.
 
-      toolbox.getTaskActionClient().submit(new SegmentNukeAction(new HashSet<>(unusedSegments)));
-
-      // Kill segments from the deep storage only if their load specs are not being used by any used segments
-      final List<DataSegment> segmentsToBeKilled = unusedSegments
-          .stream()
-          .filter(unusedSegment -> unusedSegment.getLoadSpec() == null
-                                   || !usedSegmentLoadSpecs.contains(unusedSegment.getLoadSpec()))
-          .collect(Collectors.toList());
+      final List<DataSegment> segmentsToBeKilled;
+
+      Set<DataSegment> segmentsWithUnreferencedLoadSpecs
+          = toolbox.getTaskActionClient().submit(new SegmentNukeAction(new HashSet<>(unusedSegments)));
```

Comment:
Please see my comment above.  I would suggest adding a new task action for this instead of piggybacking on the functionality of `SegmentNukeAction`, which should only nuke segments IMO.

Reply:
SegmentNukeAction has been left intact now